modules/really-core/src/main/scala/io/really/model/ReadHandler.scala

```diff
@@ -238,6 +237,9 @@ class ReadHandler(globals: ReallyGlobals) extends Actor with Stash with ActorLog
   private def getRequestFields(cmdFields: Set[FieldKey], model: Model): Set[FieldKey] =
     if (cmdFields.isEmpty) model.fields.keySet else cmdFields
 
+  private def availableFields(doc: JsObject, requestedFields: Set[FieldKey], model: Model) =
+    requestedFields.filter(f => model.fields(f).isInstanceOf[CalculatedField[_]] || (doc \ f).asOpt[JsValue].isDefined)
```

Comment:
I already wrote a helper to fetch all the reactive fields `model.reactiveFields`, not that it caches this loop so it's much better.